src/data/DataSetsD2Repository.ts

```diff
@@ -28,6 +28,16 @@ export class DataSetsD2Repository implements DataSetsRepository {
         }
     }
 
+    async post(data: object, options: object): Promise<MetadataResponse> {
+        const response = await this.api.metadata.post(data, options).getData();
+
+        if (response.status !== "OK") {
+            console.error(JSON.stringify(response.typeReports, null, 4));
+        }
```

Comment:
We have `runMetadata` (https://github.com/EyeSeeTea/d2-tools/blob/feature/datasets_copy-org-units/src/data/dhis2-utils.ts#L21). Note that, in case of an error, this function performs a rejection.

Reply:
Switched to runMetadata. Added try/catch.